src/cascadia/TerminalApp/CascadiaSettings.cpp

```diff
@@ -777,3 +803,77 @@ void CascadiaSettings::_ValidateNoDuplicateProfiles()
         _warnings.push_back(::TerminalApp::SettingsLoadWarnings::DuplicateProfile);
     }
 }
+
+// Method Description:
+// - Re-orders the list of profiles to match what the user would expect them to
+//   be. Orders profiles to be in the ordering { [profiles from user settings],
+//   [default profiles that weren't in the user profiles]}.
+// - Does not set any warnings.
+// Arguments:
+// - <none>
+// Return Value:
+// - <none>
+void CascadiaSettings::_ValidateProfilesMatchUserSettingsOrder()
+{
+    std::set<GUID, GuidOrdering> uniqueGuids{};
+    std::deque<GUID> guidOrder{};
+
+    auto collectGuids = [&](const auto& json) {
+        for (auto profileJson : _GetProfiles(json))
+        {
+            if (profileJson.isObject())
+            {
+                if (profileJson.isMember("guid"))
+                {
+                    auto guidJson{ profileJson["guid"] };
+                    auto guid = Utils::GuidFromString(GetWstringFromJson(guidJson));
+
+                    if (uniqueGuids.insert(guid).second)
+                    {
+                        guidOrder.push_back(guid);
+                    }
+                }
+            }
+        }
+    };
+
+    // Push all the userSettings profiles' GUIDS into the set
+    collectGuids(_userSettings);
+
+    // Push all the defaultSettings profiles' GUIDS into the set
+    collectGuids(_defaultSettings);
+    GuidEquality equals{};
+    // Re-order the list of _profiles to match that ordering
+    // for (gIndex=0 -> uniqueGuids.size)
+    //   pIndex = the pIndex of the profile with guid==guids[gIndex]
+    //   profiles.swap(pIndex <-> gIndex)
+    // This is O(N^2), which is kinda rough. I'm sure there's a better way
+    for (size_t gIndex = 0; gIndex < guidOrder.size(); gIndex++)
+    {
+        const auto guid = guidOrder.at(gIndex);
+        for (size_t pIndex = gIndex; pIndex < _profiles.size(); pIndex++)
+        {
+            auto profileGuid = _profiles.at(pIndex).GetGuid();

```

Comment:
If we just had a two members, a GUID->Profile map and a GUID ordering, then we'd only need to reorder the order of guids here. Instead of reordering profiles, we'd be reordering GUIDs. Fundamentally, it'd still be the same amount of work, right?

I'm open to doing that, but maybe as a follow up PR. changing the list of profiles to a map and a list of ordering spills out into TerminalPage and other places (tests) and I'd rather not pollute this PR with that change. Thoughts?